pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1116,6 +1152,173 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func runningOnKVM() bool {
+	_, enabledHVs := hypervisor.GetAvailableHypervisors()
+	for _, enabledHV := range enabledHVs {
+		if enabledHV == hypervisor.KVMHypervisorName {
+			return true
+		}
+	}
+	return false
+}
+
+func getHostCpusNum() (int, error) {
+	if !runningOnKVM() {
+		log.Warnf("Running not on KVM/QEMU system, skip getting CPUs info")
+		return 0, nil
+	}
```

Comment:
We have a function which does this already; hyper.GetHostCPUMem(). That one handles Xen and KVM so better to use it.

Reply:
Ok, I'll test it.

Reply:
Replaced